pkg/opensearch/opensearch.go

```diff
@@ -45,8 +46,124 @@ func NewOpenSearchDatasource(ctx context.Context, settings backend.DataSourceIns
+		res.Message = "Failed to parse settings"
+		return res, nil
+	}
+
+	flavor := jsonData.Get("flavor").MustString("")
+	if flavor != string(client.OpenSearch) && flavor != string(client.Elasticsearch) {
+		res.Status = backend.HealthStatusError
+		res.Message = "No version set"
+		return res, nil
+	}
+
+	_, err = client.ExtractVersion(jsonData.Get("version"))
+	if err != nil {
+		res.Status = backend.HealthStatusError
+		res.Message = "No version set"
+		return res, nil
+	}
+
+	timeField, err := jsonData.Get("timeField").String()
+	if err != nil {
+		res.Status = backend.HealthStatusError
+		res.Message = fmt.Sprintf("time field name is required, err=%v", err)
+		return res, nil
+	}
+
+	db := jsonData.Get("database").MustString()
+	indexInterval := jsonData.Get("interval").MustString()
+	ip, err := client.NewIndexPattern(indexInterval, db)
+	if err != nil {
+		res.Status = backend.HealthStatusError
+		res.Message = fmt.Sprintf("Failed to get index: %s", err)
+		return res, nil
+	}
+	// Same as the frontend check, we generate the time pattern indices for the last six hours
+	indices, err := ip.GetIndices(&backend.TimeRange{From: time.Now().Add(-6 * time.Hour), To: time.Now()})
+	if err != nil || len(indices) == 0 {
+		res.Status = backend.HealthStatusError
+		res.Message = fmt.Sprintf("Failed to get indices: %s", err)
+		return res, nil
+	}
+
+	var index string
+	var body []byte
+	// We try the indices until one successfully queries
+	for _, indexName := range indices {
+		index = indexName
+		osUrl, err := createOpensearchURL(index+"/_mapping/field/"+url.PathEscape(timeField), req.PluginContext.DataSourceInstanceSettings.URL)
+		if err != nil {
+			res.Status = backend.HealthStatusError
+			res.Message = err.Error()
+			continue
+		}
+
+		request, err := http.NewRequestWithContext(ctx, http.MethodGet, osUrl, bytes.NewBuffer(nil))
+		if err != nil {
+			res.Status = backend.HealthStatusError
+			res.Message = err.Error()
+			continue
+		}
+		request.Header = req.GetHTTPHeaders()
+
+		response, err := ds.HttpClient.Do(request)
+		if err != nil {
+			res.Status = backend.HealthStatusError
+			res.Message = err.Error()
+			continue
+		}
+
+		body, err = io.ReadAll(response.Body)
+		if err != nil {
+			res.Status = backend.HealthStatusError
+			res.Message = err.Error()
+			continue
+		}
+
+		if response.StatusCode == 200 {
+			break
+		}
+		res.Status = backend.HealthStatusError
+		res.Message = string(body)
+	}
+	if res.Status == backend.HealthStatusError {
+		return res, nil
+	}
+
+	jsonData, err = simplejson.NewJson(body)
+	if err != nil {
+		res.Status = backend.HealthStatusError
+		res.Message = fmt.Sprintf("Error parsing response: %s", err)
+		return res, nil
+	}
+	mapping, ok := jsonData.CheckGet(index)
+	if !ok {
+		res.Status = backend.HealthStatusError
+		res.Message = fmt.Sprintf("Index not found: %s", index)
+		return res, nil
+	}
+
+	timeFieldMapping, ok := mapping.Get("mappings").CheckGet(timeField)
```

Comment:
Matches the frontend behavior, which considers it a success if the index is found but the field isn't.